### ISSUE TITLE
Open project in Gandi IDE button

### ISSUE THREAD

**Issue by fath11** [ScratchAddons/ScratchAddons#6702]
### Why this would be helpful

Gandi IDE is a somewhat popular Scratch mod with an active community like Turbowarp. Since Turbowarp has its own version of this button, which inspired this addon, it is only fair if Gandi IDE could receive it, too.

### How this addon works

when you click the button that is right next to the remix button, it will redirect you to a Cocrea project page using the project's ID. e.g. `cocrea.world/projects/${project-id}`

### Possible alternatives

We could merge this with the TurboWarp Button addon. Tho you might need to rename the addon to something else.

### Additional context

![image](https://github.com/ScratchAddons/ScratchAddons/assets/124476952/b61cd07e-edb3-4b74-b416-df66b3c7d8a4)
I already made this pull request #6701 but were told to make an issue

**Comment by fath11** [comment]
More about Gandi IDE: https://getgandi.com/

**Comment by rgantzos** [comment]
1. TurboWarp doesn't have a community
2. TurboWarp is significantly more commonly used than Gandi
3. Just because TurboWarp has an addon doesn't mean Gandi needs one- in fact, it likely means the opposite; there shouldn't be an addon for every Scratch mod
4. Gandi seems to be a TurboWarp fork- when it comes to running Scratch-compatible projects, it doesn't seem to do much more than TurboWarp at all

**Comment by Secret-chest** [comment]
And let's not forget Gandi is not open source.

**Comment by Secret-chest** [comment]
Also it's Chinese

**Comment by rgantzos** [comment]
> Also it's Chinese

The site seems to be 100% English- are you only referring to the fact that the people who started it are Chinese? I'm not sure if that's a valid reason to not accept it...

**Comment by mybearworld** [comment]
> are you only referring to the fact that the people who started it are Chinese? I'm not sure if that's a valid reason to not accept it...

I agree, but it's definitely a reason to be careful.

**Comment by fath11** [comment]
> > are you only referring to the fact that the people who started it are Chinese? I'm not sure if that's a valid reason to not accept it...
> 
> I agree, but it's definitely a reason to be careful.

I get it why if you are talking about it not being open-source but if you are talking about it being made by Chinese people why do you need to be careful? they are people just like us, plus, they use servers located in the US so they are tied to their law too.

**Comment by Samq64** [comment]
I'm not sure it it's fully open source or not but https://github.com/Gandi-IDE has at least some Scratch repos on it.

Another thing to bring up is it real time calibration.

If this is added I would suggest merging it with the TurboWarp button.

**Comment by mybearworld** [comment]
> if you are talking about it being made by Chinese people why do you need to be careful? they are people just like us, plus, they use servers located in the US so they are tied to their law too.

Yes, I do not believe potato_boy or anyone working on Gandi has any malicious intentions, but I also don't think a lot of the people that make products that send all data to the Chinese government do, either. **I am not saying Gandi does that**. I'm just saying that, again, it's important to be careful about this.

**Comment by Secret-chest** [comment]
The GUI isn't there. Only the TW forks are.

**Comment by mybearworld** [comment]
> Another thing to bring up is it real time calibration.

True - this would be Scratch Addons endorsing a product that wouldn't be allowed to be linked on Scratch in an addon (outside of Scratch Addons itself, of course).

**Comment by DNin01** [comment]
> > Another thing to bring up is it real time calibration.
> 
> True - correct me if I'm wrong, but I think this would be Scratch Addons endorsing a product that wouldn't be allowed to be linked on Scratch in an addon (outside of Scratch Addons itself, of course).

We're already not allowed to talk about Scratch Addons on Scratch so this would still be okay, right?

Not saying we're going to, though.

Unless your concern is that people will start to talk about it on Scratch as a result.

**Comment by BroJac5246** [comment]
What real benefits does Gandi provide that make it better than TurboWarp? It appears to mainly be a reskined fork of it. The only big thing is real-time collaboration, which can be achieved with Blocklive.

**Comment by fath11** [comment]
> What real benefits does Gandi provide that make it better than TurboWarp? It appears to mainly be a reskined fork of it. The only big thing is real-time collaboration, which can be achieved with Blocklive.

Blocklive is not built-in. Gandi is also owned by a company called Cocrea so they have access to much more advanced servers than Turbowarp extensions for MMO, better cloud variable, and their restore point that can actually be trusted.

**Comment by fath11** [comment]
> Can Gandi projects be uploaded to the Scratch website?

Yes, but just like turbowarp if you use an extension, it cannot be uploaded to Scratch

**Comment by rgantzos** [comment]
> Blocklive is not built-in. 

Neither is Gandi 🤦‍♂️

> Gandi is also owned by a company called Cocrea so they have access to much more advanced servers than Turbowarp.

Not sure how the fact that it’s owned by a company has anything to do with being more advanced than TurboWarp in terms of servers. In addition to this, Cocrea isn’t actually a real company, at least not any more of one than TurboWarp is.

Also, another important thing to keep in mind is that users will be using this button from Scratch projects, projects that are compatible with Scratch. Any special extensions or other technologies that Gandi made up do not apply here.

**Comment by rgantzos** [comment]
Another important point is that I remember users are often encouraged to create accounts on Gandi while in the editor and on the site, which requires an email address and other personal information. Just making that clear in case it somehow would affect the Scratch Addons privacy policy.

**Comment by BroJac5246** [comment]
> > Can Gandi projects be uploaded to the Scratch website?
> 
> Yes, but just like turbowarp if you use an extension, it cannot be uploaded to Scratch

So really its only advantage is negated because the projects cannot be shared on Scratch? At that point, why bother adding this?

**Comment by CST1229** [comment]
Gandi has (or at least had) questionable privacy and other practices, according to messages by GarboMuffin from the TurboWarp Discord server:

<details><summary>click to show, there's a LOT</summary>

![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/0b7af1ea-d64b-48da-b99a-e8fea61c8a68)

![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/28294f82-32f5-4ba2-a28a-15964837d207)
![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/7d74531d-daf1-45ef-a92a-f101a62f13dd)
![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/e4458670-a284-4201-8038-d3166661e561)
![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/61832cc3-2885-42bf-8e7c-e95b63877c36)
![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/b5f0c815-c0cb-4425-9a78-c1558c92a956)
![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/b8894910-e4f9-4ca7-8cb3-2539d559be42)
![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/162cd73d-420d-41d6-aa86-0d89902f72f6)

![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/fee389a3-4e72-4a4e-8739-fd002419843e)
![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/40be681a-55d7-413d-a37d-373192f9934b)
![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/11f43ee1-b971-4ece-b968-77f91e6e8753)
![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/189ce0c9-8378-4ae9-8fbd-8d5c3a896306)
![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/8d10fbe0-a91b-489d-8524-4b9a3a7a3501)
</details>

**Comment by DNin01** [comment]
I haven't tried it, but I think it's pretty clear that there's no benefit to using Gandi to run projects as TurboWarp has pretty much everything you would need in a more privacy-protecting and open source environment. The only way I would think we would consider this is if someone came up with **a specific reason to use Gandi over TurboWarp**. Even if it's not in an addon, you could still open Scratch projects in Gandi manually, which should be sufficient for those who really need it.

Also, people aren't going to use the addon if they don't recognize what it's referring to.

An alternative would be to make an addon that allowed users to enter their own websites to open projects in, utilizing addon settings, instead of implementing pre-determined shortcuts. The addon could be called "Open In". The shortcuts could potentially be accessed through the "Copy Link" button.

**Comment by mybearworld** [comment]
> Also, people aren't going to use the addon if they don't recognize what it's referring to.

They might just enable it and be confused on what it is.

**Comment by Chiroyce1** [comment]
TurboWarp has been associated since SA for a long time and is of course trustable (run by Garbo, open source) - whereas Gandi IDE is relatively new, and as CST showed via screenshots they aren't known for their trustworthiness. The main selling point of TurboWarp is the fact that it has a compiler, which significantly speeds up the runtime speed of projects, Gandi IDE has a lot of features, which is confusing and I don't think it is worth going through every single thing on their site or every single request it makes to third parties to make sure it is safe.

**Comment by MaterArc** [comment]
if anything, PenguinMod would be a better alternative (if this were to ever get passed issue stage). For the reason that they are an open-sourced project, have many features, and also are reputable and have a fair privacy policy. But, to Chiroyce's point above, compilers are significantly more helpful and useful than a website that adds more blocks and such.

**Comment by rgantzos** [comment]
> if anything, PenguinMod would be a better alternative (if this were to ever get passed issue stage). For the reason that they are an open-sourced project, have many features, and also are reputable and have a fair privacy policy. But, to Chiroyce's point above, compilers are significantly more helpful and useful than a website that adds more blocks and such.

Penguinmod doesn't seem to add any features for Scratch-compatible projects compared to TurboWarp, apart from their change making the green flag a blue flag. The main features should be things such as compiling, interpolation, higher FPS, those types of things.

**Comment by Secret-chest** [comment]
> Penguinmod doesn't seem to add any features for Scratch-compatible projects compared to TurboWarp, apart from their change making the green flag a blue flag. The main features should be things such as compiling, interpolation, higher FPS, those types of things.

Which are in TurboWarp.

**Comment by zaid1442011** [comment]
> I haven't tried it, but I think it's pretty clear that there's no benefit to using Gandi to run projects as TurboWarp has pretty much everything you would need in a more privacy-protecting and open source environment. The only way I would think we would consider this is if someone came up with **a specific reason to use Gandi over TurboWarp**. Even if it's not in an addon, you could still open Scratch projects in Gandi manually (you could probably even do it with a bookmarklet or something), which should be sufficient for those who really need it.
> 
> Also, people aren't going to use the addon if they don't recognize what it's referring to.
> 
> An alternative would be to make an addon that allowed users to enter their own websites to open projects in, utilizing addon settings, instead of implementing pre-determined shortcuts. The addon could be called "Open In". The shortcuts could potentially be accessed through the "Copy Link" button. Not sure if we're going to do that but it's worth thinking about in this conversation.

Agree. We don't need to add every single mod users ask us to, and users can add any mod they like. I also want to recommend a feature to customize the icon of the button if the user changes from Turbowarp.
